python/cudf/cudf/core/dataframe.py

```diff
@@ -2576,6 +2576,64 @@ def insert(self, loc, name, value, nan_as_null=None):
 
         self._data.insert(name, value, loc=loc)
 
+    def diff(self, periods=1, axis=0):
+        """
+        Calculates the difference of a Dataframe element compared with
+        another element in the Dataframe, treating each column independently
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for calculating difference,
+            accepts negative values.
+        axis : {0 or ‘index’, 1 or ‘columns’}, default 0
+            Take difference over rows (0) or columns (1).
+            Only row-wise (0) shift is supported.
+
+        Returns
+        -------
+        DataFrame
+            First differences of the DataFrame.
+
+        Notes
+        -----
+        Diff currently only supports numeric dtype columns
+
+        Examples
+        --------
+        >>> import cudf
+        >>> gdf = cudf.DataFrame({'a': [1, 2, 3, 4, 5, 6],
+                                  'b': [1, 1, 2, 3, 5, 8],
+                                  'c': [1, 4, 9, 16, 25, 36]})
+        >>> gdf
+        a  b   c
```

Comment:
Missing an indentation?

Reply:
FYI, some of these issues might be caught by merging in `branch-22.02`. I just merged #9815 which will execute the doctests and help point out any errors in execution.